packages/hash/frontend/src/pages/[account-slug]/entities/[entity-id].page/entity-editor/properties-section/property-table/cells/property-name-cell.tsx

```diff
@@ -0,0 +1,124 @@
+import {
+  CustomCell,
+  CustomRenderer,
+  GridCellKind,
+} from "@glideapps/glide-data-grid";
+import {
+  getColumnPadding,
+  getYCenter,
+} from "../../../../../../../../components/GlideGlid/utils";
+import { CustomGridIcon } from "../../../../../../../../components/GlideGlid/utils/custom-grid-icons";
+import { drawCellFadeOutGradient } from "../../../../../../../../components/GlideGlid/utils/draw-cell-fade-out-gradient";
+import { drawVerticalLine } from "../../../../../../../../components/GlideGlid/utils/draw-vertical-line";
+import { TableExpandStatus } from "../../../entity-editor-context";
+import { PropertyRow } from "../types";
+
+export interface PropertyNameCellProps {
+  readonly kind: "property-name-cell";
+  property: PropertyRow;
+}
+
+export type PropertyNameCell = CustomCell<PropertyNameCellProps>;
+
+export const createRenderPropertyNameCell = (
+  togglePropertyExpand: (id: string) => void,
+  propertyExpandStatus: TableExpandStatus,
```

Comment:
Memoized custom cell renderers here https://github.com/hashintel/hash/pull/1266/commits/8b47700fd34d5ae70a8159785dd04c8270ea8824, so this'll not get re-created when parent re-renders.

Reply:
It probably doesn't matter, but to explain my thinking further – I was wondering if it mattered/caused any weird side effects for the renderer to *ever* be re-created, even if one of the variables it reads from react changes. In an ideal world, the properties could be passed into glide in a way that re-triggers a render of the glide renderers, and these will be able to read the new property and respond accordingly, without being re-created. I don't know if that's possible, or the concerns of re-created renderers even matter, but using `useMemo` only makes the problem marginally better – renderers are still being recreated when a react property changes.

Reply:
I see, renderers are just simple objects, and they can only read `cell data`. That's why I'm passing these extra props to this renderer. I don't see and issue here, since renderer is just a function, and we expect it to run the `draw` function again when the passed props are changed. 
TL;DR I don't think this will cause an issue